istioctl/pkg/kubernetes/sds_client.go

```diff
@@ -0,0 +1,147 @@
+		if isIngressGateway(agent) {
+			secretType = "gateway"
+			container = "istio-proxy"
+		}
+
+		debugOutput, err := c.NodeAgentDebugEndpointOutput(
+			agent.Name, agent.Namespace, secretType, container)
+		if err != nil {
+			return nil, err
+		}
+
+		debugResponses[agent.Name] = debugOutput
+	}
+	return debugResponses, nil
+}
+
+// NodeAgentDebugEndpointOutput makes a request to the target nodeagent's debug endpoint and returns the raw response
+func (c *Client) NodeAgentDebugEndpointOutput(podName, ns, secretType, container string) (sds.Debug, error) {
+	request := []string{
+		"curl",
+		fmt.Sprintf("%s/%s", debugEndpointPath, secretType),
+	}
+	rawResult, err := c.ExtractExecResult(podName, ns, container, request)
+	if err != nil {
+		return sds.Debug{}, err
+	}
+
+	var sdsDebug sds.Debug
+	err = json.Unmarshal(rawResult, &sdsDebug)
+	if err != nil {
+		return sds.Debug{}, fmt.Errorf("failed to unmarshal node agent debug response: %v", err)
+	}
+
+	return sdsDebug, nil
+}
+
+// nodeAgentsForPod returns all node agents which are serving secrets to the supplied pod
+// in the case of an ingress-gateway, it is possible for there to be are two corresponding nodeagents: the one on the node
+// running as a daemon set serving workload secrets, and the embedded nodeagent running in the same pod
+func (c *Client) nodeAgentsForPod(name, ns, istioNamespace string) ([]*v1.Pod, error) {
+	agentPods := make([]*v1.Pod, 0)
+
+	// need to retrieve more information on target pod, such as node and labels
+	podGet := c.Get().Resource("pods").Namespace(ns).Name(name)
+	obj, err := podGet.Do().Get()
+	if err != nil {
+		log.Debugf("failed to retrieve pod information for %s.%s: %v", ns, name, err)
+		return nil, err
+	}
+	pod, ok := obj.(*v1.Pod)
+	if !ok {
+		return nil, fmt.Errorf("failed retrieving pod: %s", name)
+	}
+
+	// if the pod is a secure ingress-gateway, we should add the pod itself, since its using an embedded nodeagent
+	if isIngressGateway(pod) && len(pod.Spec.Containers) > 1 {
+		agentPods = append(agentPods, pod)
+	}
+
+	nodeAgentFilters := map[string]string{
+		"labelSelector": "app=nodeagent",
+		"fieldSelector": "spec.nodeName=" + pod.Spec.NodeName,
+	}
+	nodeAgentPodReq := c.Get().
+		Resource("pods").
+		Namespace(istioNamespace)
+	for k, v := range nodeAgentFilters {
+		nodeAgentPodReq.Param(k, v)
+	}
+
+	res := nodeAgentPodReq.Do()
+	if res.Error() != nil {
+		log.Debugf("failed to retrieve node agent pods: %v", err)
+		return nil, fmt.Errorf("unable to retrieve node agent pods: %v", res.Error())
+	}
+	nodeAgentPods := &v1.PodList{}
+	if err := res.Into(nodeAgentPods); err != nil {
+		return nil, fmt.Errorf("unable to parse response into pod list: %v", res.Error())
+	}
+	if len(nodeAgentPods.Items) == 1 {
+		agentPods = append(agentPods, &nodeAgentPods.Items[0])
+	}
+
+	return agentPods, nil
+}
+
+func isIngressGateway(p *v1.Pod) bool {
+	appLabel, ok := p.Labels["app"]
+	if ok && appLabel == ingressGatewayApp {
```

Comment:
you can just return this expression directly

Reply:
fixed